### PR TITLE
Add fluid properties for sodium.

### LINKED ISSUE #17411: Sodium properties cannot be used in Porous flow module
## Bug Description
I'm learning moose not long ago. I don't know if it is a bug.
I can't use SodiumProperties in porous flow calculation.
I have open the porousflow module and fluid properties module in makefile.

## Steps to Reproduce
I tried to exchange the fluid property used in one example of porous flow. The address of that example is shown below.
GitHub\moose\modules\porous_flow\examples\tutorial\05.i
The original property is water. I want to use SodiumProperties instead.
So I just change the following sentence:

```
[Modules]
  [./FluidProperties]
    [./the_simple_fluid]
      type = Water97FluidProperties
    [../]
  [../]
[]
into:
[Modules]
  [./FluidProperties]
    [./the_simple_fluid]
      type = SodiumProperties
    [../]
  [../]
[]
```

But it will cause a 'Segmentation fault'.

## Impact
I don’t know if I made a mistake, because I found that it’s okay to use CO2FluidProperties.
I am a rookie, which prevents me from carrying out the next step. I hope every expert can answer it. Thank you very much.

### PR DESCRIPTION
**Pull request by aprilnovak**
Adds material properties for sodium at liquid saturation conditions. The density correlation doesn't have any dependence on pressure (which means that they cannot be used in Pronghorn because the SUPG stabilization requires derivatives of temperature with respect to internal energy and specific volume). 

I got around this issue with Flibe by adding a tiny dependence on pressure to the density correlation, but unfortunately here the density correlation is not linear and therefore can't be inverted analytically. I'm hoping that with the future FV implementation, we won't need those derivatives anyways, so this might just be a temporary limitation. If it doesn't end up being a temporary limitation, I could add a simple bisection method or something to find `dT_de` and `dT_dv` given the nonlinear `rho(T)` function.

Closes #17411

### CODE REVIEW

File: modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+#pragma GCC diagnostic ignored "-Woverloaded-virtual"
+public:
+  static InputParameters validParams();
+
+  SodiumSaturationFluidProperties(const InputParameters & parameters);
+
+  /**
+   * Fluid name
+   *
+   * @return "sodium"
```
Comment (andrsd) [inline modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h:29]: Let's do `sodium_sat` here.

File: modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+   */
+  virtual std::string fluidName() const override;
+
+  /**
+   * Density from pressure and temperature
+   *
+   * @param[in] p   pressure (Pa)
+   * @param[in] T   temperature (K)
+   * @return density (kg/m$^3$)
+   */
```
Comment (andrsd) [inline modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h:39]: You don't have to repeat the doxygen in the inherited classes. We have a flag that will use the doco from the parent if it is missing in the child. I know we talked about this before and it is nice to open a header file and see what is the meaning of all the params - especially when somebody new opens the file. I assume people would go to the parent class for this or the online doco.

This is up to you if you want to change it. I am ok either way. Just saying this is not really needed.

File: modules/fluid_properties/unit/src/SodiumSaturationFluidPropertiesTest.C [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+ * Test that the thermal conductivity and its derivatives are
+ * correctly computed.
+ */
+TEST_F(SodiumSaturationFluidPropertiesTest, thermalConductivity)
+{
+  const Real T = 800.0;
+  const Real p = 101325;
+
+  // Fink and Leibowitz (1979) list conductivity as 66.8 at 800 K, so the fit agrees well
+  REL_TEST(_fp->k_from_p_T(p, T), 66.9752096, REL_TOL_SAVED_VALUE);
```
Comment (andrsd) [inline modules/fluid_properties/unit/src/SodiumSaturationFluidPropertiesTest.C:36]: If you are comparing to a value that was computed (or measured) by something (someone) else, you should be using `REL_TOL_EXTERNAL_VALUE`. 

`REL_TOL_SAVED_VALUE` is used for: "I implemented this, I think this is the right value, so I am gonna compare to it".

There is some in-code doco in `FluidPropertiesTestUtils.h`.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by andrsd** [comment]
A few small things to fix.

Otherwise
image

File: modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h (hunk 2 above)
Comment (aprilnovak) [reply modules/fluid_properties/include/userobjects/SodiumSaturationFluidProperties.h:39]: yeah you're right, I just copied another class in here that I made a while ago before realizing that - I'll remove all the extra doxygen

File: modules/fluid_properties/unit/src/SodiumSaturationFluidPropertiesTest.C (hunk 3 above)
Comment (aprilnovak) [reply modules/fluid_properties/unit/src/SodiumSaturationFluidPropertiesTest.C:36]: I hope the comment isn't misleading - that's just there for my own sanity check. The actual values that I'm testing against were computed with a Python script where I coded up the same correlations (with the idea thinking that it's probably unlikely I code up the exact same typo 2x).

What comparison value would you recommend for that? The `REL_TOL_SAVED_VALUE` failed a bunch of the density and specific volume tests by ~1e-8.

Comment (andrsd) [reply modules/fluid_properties/unit/src/SodiumSaturationFluidPropertiesTest.C:36]: I see, you are not comparing to Fink and Lebowitz value. But, again if the value was computed by something else it should be `REL_TOL_EXTERNAL_VALUE` which is 1e-3, so fairly loose.

`REL_TOL_SAVED_VALUE` is 1e-12. That's on purpose. You can compare to the value computed by this code and use `REL_TOL_SAVED_VALUE` - there is no problem with that, but then the comments are misleading. As we witnessed :-)

**Review by andrsd — APPROVED** [review]

**Comment by andrsd** [comment]
This will be very helpful, so thanks again :+1:

**Pull request merged**